src/main/java/it/pagopa/ecommerce/commons/utils/NpgApiKeyHandler.java

```diff
@@ -0,0 +1,173 @@
+        ) {
+            addMethodPspMapping(
+                    paymentMethod,
+                    NpgPspApiKeysConfig.parseApiKeyConfiguration(pspKeys, pspToHandle, paymentMethod, objectMapper)
+                            .fold(
+                                    exception -> {
+                                        throw exception;
+                                    },
+                                    Function.identity()
+                            )
+            );
+            return this;
+        }
+
+        /**
+         * Build {@link NpgApiKeyHandler} handler instance with the given api keys
+         * configurations
+         *
+         * @return a new {@link NpgApiKeyHandler} instance or throw an error if
+         *         configuration are not correct
+         */
+        public NpgApiKeyHandler build() {
+            return new NpgApiKeyHandler(defaultApiKey, methodsApiKeyMapping);
+        }
+    }
+
+    private final String defaultApiKey;
+
+    private final Map<NpgClient.PaymentMethod, NpgPspApiKeysConfig> methodsApiKeyMapping;
+
+    /**
+     * Constructor
+     *
+     * @param defaultApiKey        NPG default api key
+     * @param methodsApiKeyMapping the payment method psp api key mapping
+     */
+    NpgApiKeyHandler(
+            String defaultApiKey,
+            Map<NpgClient.PaymentMethod, NpgPspApiKeysConfig> methodsApiKeyMapping
+    ) {
+        if (Objects.isNull(defaultApiKey)) {
+            throw new NpgApiKeyConfigurationException(
+                    "Invalid configuration detected! Default api key mapping cannot be null"
+            );
+        }
+        if (Objects.isNull(methodsApiKeyMapping) || methodsApiKeyMapping.isEmpty()) {
+            throw new NpgApiKeyConfigurationException(
+                    "Invalid configuration detected! Payment methods api key mapping cannot be null or empty"
+            );
+        }
+        this.defaultApiKey = defaultApiKey;
+        this.methodsApiKeyMapping = Collections.unmodifiableMap(methodsApiKeyMapping);
+    }
+
+    /**
+     * Get the default NPG api key
+     *
+     * @return the default api key
+     */
+    public String getDefaultApiKey() {
+        return this.defaultApiKey;
+    }
+
+    /**
+     * Get the api key associated to the input pspId for the given paymentMethod
+     *
+     * @param paymentMethod the payment method for which api keys will be searched
+     *                      for
+     * @param pspId         the searched api key psp id
+     * @return either the found api key or an NpgApiKeyConfigurationException
+     *         exception if no api key can be found
+     */
+    public Either<NpgApiKeyConfigurationException, String> getApiKeyForPaymentMethod(
+                                                                                     NpgClient.PaymentMethod paymentMethod,
+                                                                                     String pspId
+    ) {
```

Comment:
I'd add an alias called `get` to get nicer indexing when calling from Kotlin

Reply:
Fixed with https://github.com/pagopa/pagopa-ecommerce-commons/pull/188/commits/17f5fabeeddc26d002749691174e3a9fde598cf2